weasyl/controllers/messages.py

```diff
@@ -14,17 +14,18 @@
 @login_required
 @token_checked
 def messages_remove_(request):
-    form = request.web_input(recall='', remove=[])
-    remove_all_before = form.get('remove-all-before')
+    recall = request.params.get('recall', '')
+    remove = request.params.getall('remove')
+    remove_all_before = request.params.get('remove-all-before')
 
     if remove_all_before:
         message.remove_all_before(request.userid, int(remove_all_before))
-    elif form.get('remove-all-submissions'):
-        message.remove_all_submissions(request.userid, define.get_int(form['remove-all-submissions']))
+    elif request.params.get('remove-all-submissions'):
+        message.remove_all_submissions(request.userid, define.get_int(request.params.get('remove-all-submissions')))
```

Comment:
It should be safe to make this stricter by now (it always has an integer value when it’s part of the form; the leniency was for anyone who opened their form before the value for `only_before` was deployed, which was over seven years ago in the weasyl-old commit “Removing submission notifications by date”).

Normally would be changed separately, but if I’m leaving a review comment about `form[` becoming `request.params.get(` anyway… :)

```suggestion
    elif 'remove-all-submissions' in request.params:
        message.remove_all_submissions(request.userid, int(request.params['remove-all-submissions']))
```